mempool/clist_mempool.go

```diff
@@ -349,55 +362,55 @@ func (mem *CListMempool) handleCheckTxResponse(tx types.Tx, sender p2p.ID) func(
 			return
 		}
 
+		// Check that tx is not already in the mempool. This can happen when the
+		// cache overflows. See https://github.com/cometbft/cometbft/pull/890.
+		txKey := tx.Key()
+		if mem.Contains(txKey) {
+			if err := mem.addSender(txKey, sender); err != nil {
```

Comment:
the `memTx.addSender` might return `false` when a `peerID == 0` but  not sure if this is possible and what would be the implications here since in the `mem.addSender` you are just looking for a `found` and returning an error in this case. Not sure if the `!found` would be also a error case and would be possible when `peerID == 0`. 

Probably outside the scope of this PR but just commenting here since I was understanding this logic a bit better.

Reply:
`peerID == 0` handles the case when there's no sender, that is, the tx came via an RPC endpoint. It's not an error.